### PR TITLE
feature(workflow_engine): Add path for producing occurrences/status updates in `process_detectors`

### PR DESCRIPTION
**Pull request by wedamija**
This modifies the `DetectorEvaluationResult` to allow it to include an `IssueOccurrence` + `event_data` or `StatusChangeMessage`. We then use this to send occurrences to the issue platform.

### CODE REVIEW

File: src/sentry/workflow_engine/processors/detector.py [hunk 1]
```diff
@@ -58,11 +66,34 @@ def process_detectors(
                         "group_key": result.group_key,
                     },
                 )
+                continue
+
+            if result.result is not None:
+                occurrence, status_change = None, None
+                if isinstance(result.result, IssueOccurrence):
+                    occurrence = (
+                        result.result if isinstance(result.result, IssueOccurrence) else None
```
Comment (ceorourke) [inline src/sentry/workflow_engine/processors/detector.py:75]: Do you need to check `isinstance(result.result, IssueOccurrence)` twice?

File: tests/sentry/workflow_engine/processors/test_detector.py [hunk 2]
```diff
@@ -29,9 +29,7 @@ class NoHandlerGroupType(GroupType):
 
         class MockDetectorHandler(DetectorHandler[dict]):
             def evaluate(self, data_packet: DataPacket[dict]) -> list[DetectorEvaluationResult]:
```
Comment (ceorourke) [inline tests/sentry/workflow_engine/processors/test_detector.py:31]: Does this still need to take `data_packet`?

**Review by ceorourke — APPROVED** [review]
lgtm but I won't be offended if you wait for josh's approval

File: src/sentry/workflow_engine/processors/detector.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/workflow_engine/processors/detector.py:75]: Oops, left this in place from before I switched this over to an if/else

File: tests/sentry/workflow_engine/processors/test_detector.py (hunk 2 above)
Comment (wedamija) [reply tests/sentry/workflow_engine/processors/test_detector.py:31]: Yeah, evaluate always accepts only a DataPacket

File: src/sentry/workflow_engine/processors/detector.py [hunk 3]
```diff
@@ -58,11 +66,34 @@ def process_detectors(
...
+                    )
+                    payload_type = PayloadType.OCCURRENCE
+                else:
+                    status_change = result.result
+                    payload_type = PayloadType.STATUS_CHANGE
+
+                produce_occurrence_to_kafka(
```
Comment (saponifi3d) [inline src/sentry/workflow_engine/processors/detector.py:82]: 🎉 1/2 of milestone 2. heh.

**Review by saponifi3d — APPROVED** [review]

File: src/sentry/workflow_engine/processors/detector.py (hunk 3 above)
Comment (wedamija) [reply src/sentry/workflow_engine/processors/detector.py:82]: Yeah, there's wasn't much to implementing it so I just went ahead. Most of the work is producing the occurrence anyway.

File: src/sentry/workflow_engine/processors/detector.py [hunk 4]
```diff
@@ -58,11 +66,34 @@ def process_detectors(
                         "group_key": result.group_key,
                     },
                 )
+                continue
+
+            if result.result is not None:
+                occurrence, status_change = None, None
```
[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/workflow_engine/processors/detector.py:72]: Yeah, that seems reasonable

**Pull request merged**